packages/plugin-ext/src/main/browser/plugin-ext-deploy-command.ts

```diff
@@ -76,7 +80,7 @@ export class DeployQuickOpenItem extends QuickOpenItem {
 
     constructor(
         protected readonly name: string,
-        protected readonly pluginServer: PluginServer,
+        protected readonly pluginExtDeployCommandService: PluginExtDeployCommandService,
```

Comment:
I'm not sure why we want to pass ourselves (`PluginExtDeployCommandService`) in the constructor, it feels odd to me.
In addition, why aren't we just calling:

```ts
this.deployPlugin(this.name);
```

rather than

```ts
this.pluginExtDeployCommandService.deployPlugin(this.name);
```

Reply:
The method deployPlugin is not in the current this.
I've tried to pass the method in the constructor but it lost the this.
How do you suggest I can do it differently?

Reply:
@odeliat instead maybe do the customization in the creation of the item, move it to a protected method that downstream can override.
We did something very similar here: https://github.com/eclipse-theia/theia/pull/8648